### PR TITLE
PSTRESS-129 Add support for Rotation of GCache Master Key

### PR DESCRIPTION
**Pull request by mohitj1988**
https://jira.percona.com/browse/PSTRESS-129

### CODE REVIEW

**Review by saikumar-vs — APPROVED** [review]
Mohit , thanks for working on this patch, approve from testing with pstress-run.sh and pxc80.cnf file, with options
GCACHE_ENCRYPTION=0|1

o/p below 
GCACHE_ENCRYPTION=0

[20:13:52] [0] PXC Encryption run: 'YES' | GCache Encryption: 'NO'

stress/pstress/../src/pstress-pxc --database=test --threads=2 --queries-per-thread=2147483647 --logdir=/dev/shm/184147/2/ --user=root --socket=/dev/shm/184147/2/node1/node1_socket.sock --seed 184147 --step 2 --metadata-path /home/saikumar.vs/rundir/184147/ --seconds 60 --tables 10 --records 100 --log-all-queries --primary-key-probability 100 --alt-discard-tbs 0 --rotate-gcache-key 0

GCACHE_ENCRYPTION=1

[20:34:58] [0] PXC Encryption run: 'YES' | GCache Encryption: 'YES'

pstress/pstress/../src/pstress-pxc --database=test --threads=2 --queries-per-thread=2147483647 --logdir=/dev/shm/051533/1/ --user=root --socket=/dev/shm/051533/1/node1/node1_socket.sock --seed 051533 --step 1 --metadata-path /home/saikumar.vs/rundir/051533/ --seconds 60 --tables 10 --records 100 --log-all-queries --primary-key-probability 100 --alt-discard-tbs 0

the contents in the sql file after pstress execution:

Alter instance rotate gcache master key, total=>216, success=> 0
S ALTER INSTANCE ROTATE GCACHE MASTER KEY rows:0
S COMMIT rows:0

**Comment by mohitj1988** [comment]
Hi @saikumar-vs , Looks like there is something wrong. From the pasted logs, it says DDL was executed 216 times, but successful queries are 0. This does not look right. Please recheck once.

Alter instance rotate gcache master key, total=>216, success=> 0
S ALTER INSTANCE ROTATE GCACHE MASTER KEY rows:0
S COMMIT rows:0

**Comment by saikumar-vs** [comment]
> Hi @saikumar-vs , Looks like there is something wrong. From the pasted logs, it says DDL was executed 216 times, but successful queries are 0. This does not look right. Please recheck once.
> 
Hi @mohitj1988 let discussed this was just a expert from log to indicate that your patch is working and the ddl is generated.

The latest patch has been tested as well and gchace is now enabled in all nodes of pxc by pstress, thanks for explaining the feature, its good to be merged. Approved from my end.

**Review by VarunNagaraju — APPROVED** [review]

File: src/random_test.cpp [hunk 1]
```diff
@@ -248,10 +248,15 @@ int sum_of_all_options(Thd1 *thd) {
     opt_int_set(ALTER_TABLESPACE_ENCRYPTION, 0);
     opt_int_set(ALTER_MASTER_KEY, 0);
     opt_int_set(ALTER_ENCRYPTION_KEY, 0);
+    opt_int_set(ALTER_GCACHE_MASTER_KEY, 0);
     opt_int_set(ROTATE_REDO_LOG_KEY, 0);
     opt_int_set(ALTER_DATABASE_ENCRYPTION, 0);
   }
 
+  /* Disable GCache encryption for MS or PS, only supported in PXC */
+   if (strcmp(FORK, "Percona-XtraDB-Cluster") != 0)
```
Comment (rahulmalik87) [inline src/random_test.cpp:257]: better to disable it for PXC 2.4 also ?

Comment (mohitj1988) [reply src/random_test.cpp:257]: Good catch @rahulmalik87 .

Its already merged. I will push a post fix to address this. Please confirm if below fix looks good ?

  /* Disable GCache encryption for MS or PS, only supported in PXC-8.0 */
   if (strcmp(FORK, "Percona-XtraDB-Cluster") != 0 && server_version() < 80000)
      opt_int_set(ALTER_GCACHE_MASTER_KEY, 0);

Comment (mohitj1988) [reply src/random_test.cpp:257]: Good catch @rahulmalik87 . 

The PR is already merged, so I'll create a post fix and push the fix. Let me know if the below fix looks good ?
Screenshot 2022-09-21 at 11 41 26 AM

**Pull request merged**